framework/src/org/apache/cordova/engine/SystemWebChromeClient.java

```diff
@@ -211,54 +216,137 @@ public View getVideoLoadingProgressView() {
         return mVideoProgressView;
     }
 
+    // <input type=file> support:
+    // openFileChooser() is for pre KitKat and in KitKat mr1 (it's known broken in
+    // KitKat).
+    // For Lollipop, we use onShowFileChooser().
+    public void openFileChooser(ValueCallback<Uri> uploadMsg) {
+        this.openFileChooser(uploadMsg, "*/*");
+    }
+
+    public void openFileChooser(ValueCallback<Uri> uploadMsg, String acceptType) {
+        this.openFileChooser(uploadMsg, acceptType, null);
+    }
+
+    public void openFileChooser(final ValueCallback<Uri> uploadMsg, String acceptType, String capture) {
+        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+        intent.addCategory(Intent.CATEGORY_OPENABLE);
+        intent.setType("*/*");
+        parentEngine.cordova.startActivityForResult(new CordovaPlugin() {
```

Comment:
This anonymous inner-class CordovaPlugin feels sketchy to me, but I see it's the same pattern that already exists so... I guess it's okay?

If you have ideas for a better way to handle this that has less potential for weird side-effects, I'd love to hear it.

Reply:
Let's leave it for now